packages/paste-core/utilities/grid/src/Column.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {ResponsiveValue} from 'styled-system';
+import {ThemeShape} from '@twilio-paste/theme-tokens';
+import {FlexboxProps, LayoutProps, MarginProps, PaddingProps} from '@twilio-paste/style-props';
+import {Flex, FlexProps, Vertical} from '@twilio-paste/flex';
+
+type OffsetOptions = number;
+export type Offset = ResponsiveValue<OffsetOptions>;
+type SpanOptions = number;
+export type Span = ResponsiveValue<SpanOptions>;
+
+export interface ColumnProps extends FlexProps {
+  count?: number;
+  gutter?: ResponsiveValue<keyof ThemeShape['space']>;
+  offset?: Offset;
+  span?: Span;
+  vertical?: Vertical;
+}
+
+const getVertical = ({vertical}: ColumnProps): {} => {
+  if (Array.isArray(vertical)) {
+    return (vertical as Vertical[]).map((value: Vertical) => {
+      if (typeof value === 'boolean') {
+        return value === true ? '100%' : '0';
+      }
+      return null;
+    });
+  }
+
+  if (vertical) {
+    return '100%';
+  }
+
+  return '0';
+};
+
+const getSpan = ({count, span}: ColumnProps): {} => {
+  if (Array.isArray(span) && count) {
+    return (span as SpanOptions[]).map((value: SpanOptions) => {
+      return `${(value / 12) * 100}%`;
+    });
+  }
+
+  if (typeof span === 'number' && count && count <= 12) {
+    return `${(span / 12) * 100}%`;
+  }
+
+  if (count !== undefined) {
+    return `${(1 / count) * 100}%`;
+  }
+
+  return `${(1 / 12) * 100}%`;
+};
+
+const getOffset = ({offset}: ColumnProps): {} | undefined => {
+  if (Array.isArray(offset)) {
+    return (offset as OffsetOptions[]).map((value: OffsetOptions) => {
+      return `${(value / 12) * 100}%`;
+    });
+  }
+
+  if (typeof offset === 'number') {
+    return `${(offset / 12) * 100}%`;
+  }
+};
+
+const getColumnStyles = (props: ColumnProps): LayoutProps => {
+  const columnStyles: FlexboxProps & LayoutProps & MarginProps & PaddingProps = {
+    width: getSpan(props),
```

Comment:
Tried to use `Basis` here instead, but it didn't really work the `minWidth` coming from `Box`. So I passed in `width` and `minWidth` props in `Flex`.

Reply:
For `Grid`, we need to use an absolute width. `Flex-basis` would still allow the columns to grow if the content within them is wider. Basically `flex-basis` is not a guaranteed size.